lib/SILOptimizer/IPO/CapturePromotion.cpp

```diff
@@ -350,18 +350,21 @@ computeNewArgInterfaceTypes(SILFunction *F,
     assert(paramBoxTy->getLayout()->getFields().size() == 1
            && "promoting compound box not implemented yet");
     auto paramBoxedTy = paramBoxTy->getFieldType(F->getModule(), 0);
-    auto &paramTL = F->getModule().Types.getTypeLowering(paramBoxedTy);
-
-    ParameterConvention convention;
-    if (paramTL.isFormallyPassedIndirectly()) {
-      convention = ParameterConvention::Indirect_In;
-    } else if (paramTL.isTrivial()) {
-      convention = ParameterConvention::Direct_Unowned;
-    } else {
-      convention = ParameterConvention::Direct_Owned;
+    {
+      Lowering::GenericContextScope GenericScope(
+          F->getModule().Types, paramBoxTy->getLayout()->getGenericSignature());
+      auto &paramTL = F->getModule().Types.getTypeLowering(paramBoxedTy);
```

Comment:
The box knows how to lower itself, yes. But it returns a lowered SILType.
And need this `TypeLowering paramTL` , because the code that follows uses it in conditional statements like:
```
      if (paramTL.isFormallyPassedIndirectly()) {
        convention = ParameterConvention::Indirect_In;
      } else if (paramTL.isTrivial()) {
        convention = ParameterConvention::Direct_Unowned;
      } else {
        convention = ParameterConvention::Direct_Owned;
```
I'm not sure how to call `isFormallyPassedIndirectly` directly, because it asks for a `GenericSignature` and `ResilienceExpansion`.